src/status_im/contexts/wallet/effects.cljs

```diff
@@ -46,6 +46,27 @@
                         (when (and error (fn? on-error))
                           (on-error error)))))))
 
+(defn create-account-from-private-key
+  [private-key]
+  (-> private-key
+      (security/safe-unmask-data)
+      (native-module/create-account-from-private-key)
+      (promesa/then (fn [result]
+                      (let [{:keys [address emojiHash keyUid
+                                    publicKey privateKey]} (transforms/json->clj result)]
```

Comment:
When dealing with primitive values (these are all strings?), it's a bit better to use `oops/oget` and then we can skip the `json->clj` transformation here.